### PR TITLE
[NCL-8044] Add git backend switching

### PR DESCRIPTION
**Pull request by pkocandr**
### All Submissions:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/repour/wiki/Changelog) for your change?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by pkocandr** [comment]
retest this please

**Comment by pkocandr** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by pkocandr** [comment]
retest this please

File: repour/server/endpoint/internal_scm_gitlab.py [hunk 1]
```diff
@@ -82,6 +71,11 @@ async def internal_scm_gitlab(spec, repo_provider):
     return result
 
 
+def read_token(token_filepath):
+    with open(token_filepath, "r") as token_file:
+        return token_file.read()
```
Comment (thescouser89) [inline repour/server/endpoint/internal_scm_gitlab.py:76]: You might want to do:
```python
return token_file.read().strip()
```

to remove any newline at the end of the token when read in the file.

**Review by thescouser89 — CHANGES_REQUESTED** [review]

File: repour/server/endpoint/internal_scm_gitlab.py (hunk 1 above)
Comment (thescouser89) [reply repour/server/endpoint/internal_scm_gitlab.py:76]: meh if that's the only issue I could find, I'll just merge it and create another PR for it.

**Review by thescouser89 — APPROVED** [review]

**Pull request merged**